sdks/java/core/src/main/java/org/apache/beam/sdk/transforms/DoFnTester.java

```diff
@@ -51,56 +53,37 @@
 import org.apache.beam.sdk.values.TupleTag;
 import org.apache.beam.sdk.values.ValueInSingleWindow;
 import org.joda.time.Instant;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
 
 /**
- * A harness for unit-testing a {@link DoFn}.
- *
- * <p>For example:
- *
- * <pre> {@code
- * DoFn<InputT, OutputT> fn = ...;
- *
- * DoFnTester<InputT, OutputT> fnTester = DoFnTester.of(fn);
- *
- * // Set arguments shared across all bundles:
- * fnTester.setSideInputs(...);      // If fn takes side inputs.
- * fnTester.setOutputTags(...);  // If fn writes to more than one output.
- *
- * // Process a bundle containing a single input element:
- * Input testInput = ...;
- * List<OutputT> testOutputs = fnTester.processBundle(testInput);
- * Assert.assertThat(testOutputs, Matchers.hasItems(...));
- *
- * // Process a bigger bundle:
- * Assert.assertThat(fnTester.processBundle(i1, i2, ...), Matchers.hasItems(...));
- * } </pre>
- *
- * @param <InputT> the type of the {@link DoFn}'s (main) input elements
- * @param <OutputT> the type of the {@link DoFn}'s (main) output elements
+ * @deprecated Instead of this, write a {@link TestPipeline} that verifies expectations using {@link
+ *     PAssert} and run it with the {@code DirectRunner} (or other runners, as applicable). For
+ *     simple cases, use {@link Create} to provide input. For complex cases involving triggers, use
+ *     {@link TestStream} for input data and behavior.
```

Comment:
I'm actually not sure if TestStream is relevant at all to testing of DoFn's. What TestStream simulates, affects GBKs and Combine's, not ParDo's.

Reply:
It does enforce bundle boundaries, though I'm not sure it is required to do so.

Reply:
Hm it's an interesting point, but I'm not sure of that either; or at least I've never seen a test that relies on that, so it's not a common test pattern (testing for particular cases of bundle boundaries is also pretty rare) and I still would suggest to recommend people use TestPipeline. If it turns out that TestStream is useful for testing bundle boundaries, then make a note about that too?

Reply:
I didn't port IOs that use `DoFnTester`. I know they are interested in bundle boundaries. I've moved it to the `DoFn` javadoc and rephrased to be quite explicit. WDYT?